specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -84,45 +86,46 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
           fieldName as string
         )
       );
-
-    Promise.all(checks)
-      .then((results) => {
-        return thisCheck === this.fieldChangePromises[fieldName as string]
-          ? this.processCheckFieldResults(fieldName, results)
-          : undefined;
-      })
-      .then(() => thisCheck.resolve('finished'));
+    return Promise.all(checks).then((results) =>
+      thisCheck === this.fieldChangePromises[fieldName as string]
+        ? this.processCheckFieldResults(fieldName, results)
+        : [{ valid: true }]
+    );
   }
 
   private processCheckFieldResults(
     fieldName: keyof SCHEMA['fields'],
     results: RA<BusinessRuleResult<SCHEMA> | undefined>
-  ) {
-    results.map((result) => {
-      if (result !== undefined) {
-        if (result.key === undefined) {
-          if (result.valid && typeof result.action === 'function') {
-            result.action();
-          }
-        } else if (result.valid === false) {
-          this.resource.saveBlockers!.add(
-            result.key,
-            fieldName as string,
-            result.reason
-          );
-        } else {
-          this.resource.saveBlockers!.remove(result.key);
-          if (typeof result.action === 'function') {
-            result.action();
+  ): RA<BusinessRuleResult<SCHEMA>> {
+    return filterArray(
+      results.map((result) => {
+        if (result !== undefined) {
```

Comment:
(optional) I know that this code was this way before, but you could make it more readable if you want

instead of 

```ts
if(result === undefined) { ... }
return undefined;
```
do

```ts
if (result === undefined) return undefined;
// ...
```

this is a matter of preference, but I prefer to return as soon as possible, as that makes code less indented (this currently looks like a [pyramid of doom](https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming)), with 3 nested if statements
plus, this gets rid of a negated condition (negation adds needless cognitive overhead)

Reply:
Yes, a guard clause here would look sooo much cleaner!
Fixed in https://github.com/specify/specify7/pull/3792/commits/7a9d0b79016dad7adeb95f333e41479290935b13